include/readout/FrameErrorRegistry.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file RawWIBTp.hpp Error registry for missing frames
+ *
+ * This is part of the DUNE DAQ , copyright 2021.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+#ifndef READOUT_INCLUDE_READOUT_FRAMEERRORREGISTRY_HPP_
+#define READOUT_INCLUDE_READOUT_FRAMEERRORREGISTRY_HPP_
+
+#include <deque>
+
+namespace dunedaq {
+namespace readout {
+
+class FrameErrorRegistry
+{
+public:
+  struct FrameError
+  {
+  public:
+    FrameError(uint64_t start_ts, uint64_t end_ts) // NOLINT(build/unsigned)
+      : start_ts(start_ts)
+      , end_ts(end_ts)
+    {}
+
+    uint64_t start_ts; // NOLINT(build/unsigned)
+    uint64_t end_ts; // NOLINT(build/unsigned)
+
+    bool operator<(const FrameError& other) const { return end_ts < other.end_ts; }
+
+    bool operator>(const FrameError& other) const { return end_ts > other.end_ts; }
+  };
+
+  FrameErrorRegistry()
+    : m_errors()
+  {}
+
+  void add_error(FrameError error) { m_errors.push_back(error); }
+
+  void update_latest_frame_in_buffer(uint64_t ts) // NOLINT(build/unsigned)
```

Comment:
This function signature is a bit confusing. In which buffer? Updating latest frame? 
Here we pop from the "current errors". Anything like "remove errors until(timestamp)" sounds better.